proc/compiler/compiler.go

```diff
@@ -0,0 +1,163 @@
+package compiler
+
+import (
+	"fmt"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/expr"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/proc/cut"
+	filterproc "github.com/brimsec/zq/proc/filter"
+	"github.com/brimsec/zq/proc/groupby"
+	"github.com/brimsec/zq/proc/head"
+	"github.com/brimsec/zq/proc/merge"
+	"github.com/brimsec/zq/proc/orderedmerge"
+	"github.com/brimsec/zq/proc/pass"
+	"github.com/brimsec/zq/proc/put"
+	"github.com/brimsec/zq/proc/rename"
+	"github.com/brimsec/zq/proc/sort"
+	"github.com/brimsec/zq/proc/split"
+	"github.com/brimsec/zq/proc/tail"
+	"github.com/brimsec/zq/proc/top"
+	"github.com/brimsec/zq/proc/uniq"
+)
+
+type Custom interface {
+	Compile(ast.Proc, proc.Parent) (proc.Interface, error)
+}
```

Comment:
Will do, but I was thinking the indexer AST extensions should just be integrated into proc and we could get rid of the custom compiler stuff altogether.  This would come in a later PR.